File: src/main/java/com/natia/secretmod/utils/AsyncAwait.java
package com.natia.secretmod.utils;

import com.google.common.base.Stopwatch;

import java.util.concurrent.TimeUnit;

public class AsyncAwait {
    public static void start(Runnable run, int millis) {
        new Thread(() -> {
            try {
                Thread.sleep(millis);
                run.run();
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }).start();
    }

    public static void until(Runnable run, int seconds, Stopwatch stopwatch) {
        if (!stopwatch.isRunning()) stopwatch.start();

        if (stopwatch.elapsed(TimeUnit.SECONDS) >= seconds) {
            stopwatch.reset();
            run.run();
        }
    }
}

File: src/main/kotlin/natia/skytweaks/utils/AsyncAwait.kt
package natia.skytweaks.utils

import com.google.common.base.Stopwatch

import java.util.concurrent.TimeUnit

object AsyncAwait {
    fun start(run: () -> Unit, millis: Int) {
        Thread {
            try {
                Thread.sleep(millis.toLong())
                run()
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
        }.start()
    }

    fun until(run: () -> Unit, seconds: Int, stopwatch: Stopwatch) {
        if (!stopwatch.isRunning) stopwatch.start()

        if (stopwatch.elapsed(TimeUnit.SECONDS) >= seconds) {
            stopwatch.reset()
            run()
        }
    }
}
